Neos.Flow/Classes/Http/ContentStream.php

```diff
@@ -0,0 +1,314 @@
+        }
+
+        $result = fwrite($this->resource, $string);
+
+        if (false === $result) {
+            throw new \RuntimeException('Error writing to stream', 1453892247);
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isReadable()
+    {
+        if (!$this->hasValidResource()) {
+            return false;
+        }
+
+        $meta = stream_get_meta_data($this->resource);
+        $mode = $meta['mode'];
+
+        return (strstr($mode, 'r') || strstr($mode, '+'));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read($length)
+    {
+        $this->ensureResourceReadable();
+
+        $result = fread($this->resource, $length);
+
+        if ($result === false) {
+            throw new \RuntimeException('Error reading stream', 1453892260);
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContents()
+    {
+        $this->ensureResourceReadable();
+
+        $result = stream_get_contents($this->resource);
+        if ($result === false) {
+            throw new \RuntimeException('Error reading from stream', 1453892266);
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMetadata($key = null)
+    {
+        if ($key === null) {
+            return stream_get_meta_data($this->resource);
+        }
+
+        $metadata = stream_get_meta_data($this->resource);
+        if (!array_key_exists($key, $metadata)) {
+            return null;
+        }
+
+        return $metadata[$key];
+    }
+
+    /**
+     * Set the internal stream resource.
+     *
+     * @param string $stream String stream target or stream resource.
+     * @param string $mode Resource mode for stream target.
+     * @return void
+     */
+    protected function openStream($stream, $mode = 'r')
+    {
+        $resource = fopen($stream, $mode);
+        $this->resource = $resource;
+    }
+
+    protected function ensureResourceReadable()
+    {
+        if ($this->isReadable() === false) {
+            throw new \RuntimeException('Stream is not readable.', 1453892039);
+        }
+    }
+
+    /**
+     *
+     */
+    protected function ensureResourceOpen()
+    {
+        if (!$this->hasValidResource()) {
+            throw new \RuntimeException('No resource available to apply operation', 1453891806);
+        }
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function hasValidResource()
+    {
+        return (is_resource($this->resource) && get_resource_type($this->resource) === 'stream');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        if (!$this->isReadable()) {
+            return '';
+        }
+
+        try {
+            $this->rewind();
+
+            return $this->getContents();
+        } catch (\Exception $e) {
+            return '';
```

Comment:
Would mean to inject a logger into this class. I really would like to avoid that. But we can ofc change that if there is a general agreement for this.